packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -0,0 +1,289 @@
+import * as React from 'react';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { useComposedRefs } from '@radix-ui/react-compose-refs';
+import { createContext } from '@radix-ui/react-context';
+import { useControllableState } from '@radix-ui/react-use-controllable-state';
+import * as PopperPrimitive from '@radix-ui/react-popper';
+import { Portal } from '@radix-ui/react-portal';
+import { Presence } from '@radix-ui/react-presence';
+import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
+import { useId } from '@radix-ui/react-id';
+import { Slot } from '@radix-ui/react-slot';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+
+/* -------------------------------------------------------------------------------------------------
+ * HoverCard
+ * -----------------------------------------------------------------------------------------------*/
+
+const HOVERCARD_NAME = 'HoverCard';
+
+type HoverCardContextValue = {
+  triggerRef: React.RefObject<HTMLAnchorElement>;
+  contentId: string;
+  open: boolean;
+  onOpenChange(open: boolean): void;
+  onMouseEnter(): void;
+  onMouseLeave(): void;
+  hasCustomAnchor: boolean;
+  onCustomAnchorAdd(): void;
+  onCustomAnchorRemove(): void;
+};
+
+const [HoverCardProvider, useHoverCardContext] = createContext<HoverCardContextValue>(
+  HOVERCARD_NAME
+);
+
+type HoverCardOwnProps = {
+  open?: boolean;
+  defaultOpen?: boolean;
+  enterDelayDuration?: number;
+  exitDelayDuration?: number;
+  onOpenChange?: (open: boolean) => void;
+};
+
+const HoverCard: React.FC<HoverCardOwnProps> = (props) => {
+  const {
+    children,
+    open: openProp,
+    defaultOpen,
+    onOpenChange,
+    enterDelayDuration = 700,
+    exitDelayDuration = 400,
+  } = props;
+  const triggerRef = React.useRef<HTMLAnchorElement>(null);
+  const [hasCustomAnchor, setHasCustomAnchor] = React.useState(false);
+  const enterEnvoked = React.useRef(false);
+  const enterTimerRef = React.useRef(0);
+  const leaveTimerRef = React.useRef(0);
+
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  const clearEnterTimer = React.useCallback(() => {
+    clearTimeout(enterTimerRef.current);
+  }, [enterTimerRef]);
```

Comment:
`enterTimerRef` doesn't need to be a dependency because ref references don't change over time.

Reply:
We could probably even do away from creating these 2 functions and just inline them as they do not necessitate any extra dependencies and arguably

`clearLeaveTimer` doesn't convey that much more intent that `clearTimeout(leaveTimerRef.current)`.

Just a suggestion, I'll let you test it out and see what you think.

Reply:
Nice spot on the deps, I agree with your second point too.

Reply:
I agree with Benoit here, it would be nice to just inline the `clearTimeout()` where used to reduce some boilerplate since the abstraction doesn't convey additional meaning. It feels a bit like a premature application of the DRY principle, sowwie 🙈